src/Microsoft.AspNet.Razor/TagHelpers/TagHelperDescriptorProvider.cs

```diff
@@ -72,11 +80,21 @@ private void Register(TagHelperDescriptor descriptor)
         {
             HashSet<TagHelperDescriptor> descriptorSet;
 
-            // Ensure there's a List to add the descriptor to.
-            if (!_registrations.TryGetValue(descriptor.TagName, out descriptorSet))
+            if (_tagHelperPrefix == null)
+            {
+                _tagHelperPrefix = descriptor.Prefix;
+            }
+
+            var registrationKey = 
+                string.Equals(descriptor.TagName, CatchAllDescriptorTarget, StringComparison.Ordinal) ? 
+                CatchAllDescriptorTarget : 
+                descriptor.FullTagName;
```

Comment:
nit: trailing whitespace.  suggest installing the trailing whitespace visualizer (thx @madskristensen !)